sources/api/apiserver/src/server/controller.rs

```diff
@@ -863,12 +1155,173 @@ mod test {
+            "\"once\"",
+            &Committed::Live,
+        )
+        .unwrap();
+        let mut result = HashMap::new();
+        let raw_setting_generator = RawSettingsGenerator {
+            command: "\"json string\"".to_owned(),
+            depth: 1,
+            strength: Strength::Weak,
+        };
+
+        let setting_generator = SettingsGenerator::from(raw_setting_generator.clone());
+
+        expand_setting_generator(&ds, &data_key, raw_setting_generator, &mut result).unwrap();
+
+        let expected = hashmap!("settings.abc.xyz.def".to_owned() => serde_json::to_value(setting_generator).unwrap());
+        assert_eq!(result, expected);
+    }
+
+    #[test]
+    fn expand_setting_generator_with_depth_zero() {
+        let mut ds = MemoryDataStore::new();
+        let data_key = Key::new(KeyType::Data, "settings.abc.def").unwrap();
+        ds.set_metadata(
+            &Key::new(KeyType::Meta, "my-meta").unwrap(),
+            &data_key,
+            "\"json string\"",
+            &Committed::Live,
+        )
+        .unwrap();
+        ds.set_key(
+            &Key::new(KeyType::Data, "settings.abc.xyz.mode").unwrap(),
+            "\"once\"",
+            &Committed::Live,
+        )
+        .unwrap();
+        let mut result = HashMap::new();
+        let raw_setting_generator = RawSettingsGenerator {
+            command: "\"json string\"".to_owned(),
+            depth: 0,
+            strength: Strength::Weak,
+        };
+
+        let setting_generator = SettingsGenerator::from(raw_setting_generator.clone());
+
+        expand_setting_generator(&ds, &data_key, raw_setting_generator, &mut result).unwrap();
+
+        let expected = hashmap!("settings.abc.def".to_owned() => serde_json::to_value(setting_generator).unwrap());
+        assert_eq!(result, expected);
+    }
+
+    #[test]
+    fn test_child_key_len() {
+        let mut ds = MemoryDataStore::new();
+        let data_key = Key::new(KeyType::Data, "settings.abc.def").unwrap();
+        ds.set_metadata(
+            &Key::new(KeyType::Meta, "my-meta").unwrap(),
+            &data_key,
+            "\"json string\"",
+            &Committed::Live,
+        )
+        .unwrap();
+
+        ds.set_key(
+            &Key::new(KeyType::Data, "settings.abc.def").unwrap(),
+            "\"once\"",
+            &Committed::Live,
+        )
+        .unwrap();
+
+        let mut result = HashMap::new();
+        let raw_setting_generator = RawSettingsGenerator {
+            command: "\"json string\"".to_owned(),
+            depth: 1,
+            strength: Strength::Weak,
+        };
+
+        expand_setting_generator(&ds, &data_key, raw_setting_generator, &mut result).unwrap();
+
+        for key_name in result.keys() {
+            let key = Key::new(KeyType::Data, key_name).unwrap();
+            assert_eq!(key.segments().len(), 4);
+        }
+    }
+
+    #[test]
+    fn test_dynamic_setting_is_only_applied_on_parent() {
```

Comment:
nit: maybe this or `child_maps`
```suggestion
    fn test_dynamic_setting_is_only_applied_on_descendants() {
```